projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -139,13 +143,15 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
           const values: MapValue[] = [];
 
           for (let i = 0; i < control.length; i++) {
-            if (filterControl(control.at(i), key)) {
+            if (filterControl(control.at(i), key, true)) {
               values.push(mapControl(control.at(i), key));
             }
           }
 
-          controls[key] = values;
-        } else if (control && filterControl(control, key)) {
+          if (values.length > 0 && !values.every(x => isNullOrUndefined(x))) {
```

Comment:
Array.prototype.some is marginally faster than `!every`

Reply:
also easier to read :)

Reply:
Yep good catch. Updated